user-console/src/main/java/org/pentaho/mantle/client/ui/tabs/MantleTabPanel.java

```diff
@@ -58,12 +66,75 @@ public class MantleTabPanel extends org.pentaho.gwt.widgets.client.tabs.PentahoT
   private static final String FRAME_ID_PRE = "frame_"; //$NON-NLS-1$
   private static int frameIdCount = 0;
 
+  private static MenuBar tabsMenuBar;
+  private static MenuItem tabsMenuItem;
+  private static TabContextMenuBar tabsSubMenuBar;
+
+  private static final String EMPTY_TABS_MENU = "empty-tabs-menu";
+
+  private HashMap<PentahoTab, MantleTabMenuItem> menuItemHashMap;
+
+  private static class TabContextMenuBar extends com.google.gwt.user.client.ui.MenuBar {
+
+    String height, maxHeight;
+    int left, top, menuBarTop;
+
+    public TabContextMenuBar( boolean b ) {
+      super( b );
+    }
+
+    @Override
+    public void onBrowserEvent( Event event ) {
+      super.onBrowserEvent( event );
+
+      MenuItem item = MenuBarUtils.findItem( this, DOM.eventGetTarget( event ) );
+      int type = event.getTypeInt();
+      switch ( type ) {
+        case Event.ONMOUSEOVER: {
+          if ( item != null ) {
+            DecoratedPopupPanel popup = MenuBarUtils.getPopup( this );
+            left = this.getOffsetWidth();
+            top = item.getAbsoluteTop();
+            menuBarTop = this.getAbsoluteTop();
+            if ( top < menuBarTop ) {
+              top = menuBarTop + 1;
+            }
+            height = "calc( 100vh - " + top + "px )";
```

Comment:
magic number/ magic calculation, add comment to briefly explain what is going on here.